### PR TITLE
Add support for app-level test_info

### PR DESCRIPTION
**Pull request by mpirri**
So that we can set beta_app_feedback_email and beta_app_description

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

Add support for setting TestFlight app-level test info. Previously we had tried to set the beta feedback email and app description through build-level test-info, but this does not work. As such, when specifying feedback email and app description through `pilot`, we now set the app-level test info.

### CODE REVIEW

File: spaceship/lib/spaceship/test_flight/app_test_info.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module Spaceship::TestFlight
+  class AppTestInfo < Base
+    # AppTestInfo wraps a test_info and beta_review_info in the format required to manage test_info
+    # for an application. Note that this structure, although looking similar to build test_info
+    # is test information about the application
+
+    attr_accessor :test_info
```
Comment (ohayon) [inline spaceship/lib/spaceship/test_flight/app_test_info.rb]: Why does a test info object have a test_info attribute?

**Review by ohayon — APPROVED** [review]

**Comment by KrauseFx** [comment]
Looks awesome. What's the easiest way to test it on a production system? As part of _pilot_, right?

**Comment by mpirri** [comment]
@KrauseFx I wrote a RakeFile to test it, but yeah, easiest way is to run from `pilot`. Automated testing in `pilot` is minimal.  😢

File: pilot/lib/pilot/build_manager.rb [hunk 2]
```diff
@@ -51,16 +51,24 @@ def distribute(options, build: nil)
...
+        end
+      end
+
+      if should_update_build_information(options)
+        begin
+          build.update_build_information!(whats_new: options[:changelog])
+        rescue => ex
+          UI.user_error!("Could not set changelog: #{ex}")
+        end
+        UI.success "Successfully set the changelog for build"
```
Comment (ohayon) [inline pilot/lib/pilot/build_manager.rb]: I think this might be nicer in the `begin` block.

File: pilot/lib/pilot/build_manager.rb [hunk 3]
```diff
@@ -51,16 +51,24 @@ def distribute(options, build: nil)
...
-        UI.success "Successfully set the changelog for build"
-      end
-
       if should_update_app_test_information(options)
         app_test_info = Spaceship::TestFlight::AppTestInfo.find(app_id: build.app_id)
         app_test_info.test_info.feedback_email = options[:beta_app_feedback_email] if options[:beta_app_feedback_email]
         app_test_info.test_info.description = options[:beta_app_description] if options[:beta_app_description]
-        app_test_info.save_for_app(app_id: build.app_id)
+        begin
+          app_test_info.save_for_app(app_id: build.app_id)
```
Comment (ohayon) [inline pilot/lib/pilot/build_manager.rb]: Should we put a `UI.success` call here with some nice message?

**Review by ohayon — APPROVED** [review]
Two tiny little things but otherwise 🚀

**Review by ohayon — APPROVED** [review]
🐑

File: spaceship/spec/test_flight/app_test_info_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+                                            'betaReviewInfo' => {
+                                              'contactFirstName' => 'First',
+                                              'contactLastName' => 'Last',
+                                              'contactPhone' => '1234567890',
+                                              'contactEmail' => 'contact@email.com',
+                                              'demoAccountName' => 'User Name',
+                                              'demoAccountPassword' => 'Password',
+                                              'demoAccountRequired' => false,
+                                              'notes' => 'notes!!'
+                                            }
```
Comment (jamesbebbington) [inline spaceship/spec/test_flight/app_test_info_spec.rb]: Could the pilot docs please be updated to show an example of adding these settings to your `Fastfile` (or however they meant to be set). Thanks.

Comment (jamesbebbington) [inline spaceship/spec/test_flight/app_test_info_spec.rb]: Ah, scratch that, I [found them](https://github.com/fastlane/fastlane/blob/master/deliver/Deliverfile.md#app_review_information).

**Pull request merged**